onnxruntime/python/onnxruntime_pybind_state_common.h

```diff
@@ -507,5 +507,6 @@ std::shared_ptr<IExecutionProviderFactory> CreateExecutionProviderFactory_Nnapi(
 std::shared_ptr<IExecutionProviderFactory> CreateExecutionProviderFactory_Rknpu();
 std::shared_ptr<IExecutionProviderFactory> CreateExecutionProviderFactory_CoreML(uint32_t flags);
 
+std::vector<std::shared_ptr<IExecutionProvider>> GetExecutionProviders(int device_type, int device_id);
```

Comment:
this function does not need to be declared here. Please define it in an unnamed namespace inside orttraining/orttraining/python/orttraining_pybind_state.cc